get_class_matched_articles.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import os
+import re
+import shutil
+import requests
+import sys
+
+class DBpediaSearcher:
+    """
+    Run query to Virtuoso SPARQL endpoint
+    """
+    
+    def _get_dbpedia_url(self, language):
+        return u'http://' + ('' if language == u'en' else language + u'.') + u'dbpedia.org'
+
+    def query_endpoint(self, query, language):
+        """
+        Run a query to Virtuoso SPARQL endpoint with given language.
+        Returns results when success, otherwise error stutus and message
+        """
+        params = {'query': query, 'format': 'json'}
+        r = requests.get(self._get_dbpedia_url(language)+'/sparql', params=params)
+        return r.json()['results']['bindings'] if r.ok else {'id_error': r.status_code, 'message': r.reason}
+
+class ArticleExtractor:
+    """
+    Load Wikipedia articles with specific language and ontology class on DBpedia
+    """
+    
+    def __init__(self, debug=True):
+        self.searcher = DBPediaSearcher()
+        self.debug = debug
+
+    def _get_article_ids(language, ontology_class):
+        query = """\
+        SELECT ?id WHERE { 
+          ?s a <http://dbpedia.org/ontology/%s> ;
+          <http://dbpedia.org/ontology/wikiPageID> ?id .
+        }\
+        """ % (ontology_class) 
+        r = self.searcher.run_query(qury, language)
+        if r.has_key(id_error):
+            print 'ERROR: Cannot access SPARQL endpoint'
```

Comment:
You are losing specific error codes from the server here. You should just print what `query_endpoint` returns